src/dsp_tools/models/input_error.py

```diff
@@ -0,0 +1,131 @@
+
+# pylint: disable=too-few-public-methods
+
+
+separator = "\n\t"
+list_separator = "\n\t- "
+
+
+class Problem(Protocol):
+    """Information about input errors."""
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+
+
+@dataclass(frozen=True)
+class DockerNotStartedProblem:
+    """This class contains information if the docker program is not started."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class DirectoryProblem:
+    """This class contains information if a path does not exist."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class FileNotFoundProblem:
+    """This class contains information if a file is not found at the expected location."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class ExcelStructureProblem:
+    """This class contains information if the structure (column names, etc.) is incorrect."""
+
+    user_msg: str
+    column: list[str] | None = None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+        msg = self._generate_error_msg()
+        return msg
+
+    def _generate_error_msg(self) -> str:
+        msg = self.user_msg + separator
+        if self.column:
+            msg += "Column(s):" + list_separator.join(self.column)
+        return msg
+
+
+@dataclass(frozen=True)
+class ExcelContentProblem:
+    """This class contains information if the content of an Excel is not as expected."""
+
+    user_msg: str
+    column: str | None = None
+    rows: list[int] | None = None
+    values: list[str] | None = None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+        msg = self._generate_error_msg()
+        return msg
+
+    def _generate_error_msg(self) -> str:
+        msg = [self.user_msg]
+        if self.column:
+            msg.append("Column: " + self.column)
+        if self.rows:
+            rws = [str(x) for x in self.rows]
+            msg.append("Row(s):" + list_separator + list_separator.join(rws))
+        if self.values:
+            msg.append("Value(s):" + list_separator + list_separator.join(self.values))
+        return separator.join(msg)
+
+
+@dataclass(frozen=True)
+class JsonValidationProblem:
+    """This class contains information about a JSON that fails its validation against the schema."""
+
+    user_msg: str
+    original_msg: str | None = None
+    message_path: str | None = None
+    property: str | None = None
+    resource_class: str | None = None
+    excel_column: str | None = None
+    excel_row: int | None = None
```

Comment:
these signatures suggest to me that there is room for improvement here 😉   
are some of those mutually exclusive? in that case, it might be nicer to store them as unions in the same field.